### PR TITLE
Pod & container level enhancements

### PR DESCRIPTION
**Pull request by rolikoff**
1. This PR adds support for new pod and container level fields that are supported by StackPath API.

Full list of new fields are: 
**Instance level**
- initContainers
- hostAliases
- dnsConfig
- shareProcessNamespace
- runAsUser
- runAsGroup
- runAsNonRoot
- supplementalGroups
- sysctls

**Container's level**
- imagePullPolicy
- args
- workingDir
- lifecycle
- terminationMessagePath
- terminationMessagePolicy
- startupProbe
- terminationGracePeriodSeconds
- runAsUser
- runAsGroup
- runAsNonRoot
- allowPrivilegeEscalation
- capabilities

2.  Adds exec action support for probes and lifecycle hooks
3. Adds new SP-6, SP-7, SP-8 geometries support.

### CODE REVIEW

**Review by denis-adamcho — APPROVED** [review]
LGTM

File: internal/provider/workload_translation.go [hunk 1]
```diff
@@ -63,15 +64,67 @@ func (p *StackpathProvider) getWorkloadSpecFrom(pod *v1.Pod) (*workload_models.V
...
 
+func (p *StackpathProvider) getWorkloadRuntimeSettingsFrom(spec v1.PodSpec) *workload_models.V1WorkloadInstanceRuntimeSettings {
+
+	runtime := workload_models.V1WorkloadInstanceRuntimeSettings{}
+	settings := workload_models.V1WorkloadInstanceContainerRuntimeSettings{}
+	settingsExist := false
+
+	if spec.TerminationGracePeriodSeconds != nil {
+		settingsExist = true
+		settings.TerminationGracePeriodSeconds = strconv.FormatInt(int64(*spec.TerminationGracePeriodSeconds), 10)
```
Comment (shayrybak) [inline internal/provider/workload_translation.go:93]: can this operation fail? if so should we handle that error case?

File: internal/provider/workload_translation.go [hunk 2]
```diff
@@ -186,29 +243,136 @@ func (p *StackpathProvider) getWorkloadContainerSpecFrom(k8sContainer *v1.Contai
...
+	return &lifecycle, nil
+}
+
+func (p *StackpathProvider) getContainerLifecycleHandlerFrom(k8sLifecycleHandler *v1.LifecycleHandler, k8sPorts []v1.ContainerPort) (*workload_models.V1ContainerLifecycleHandler, error) {
+	if k8sLifecycleHandler == nil || *k8sLifecycleHandler == (v1.LifecycleHandler{}) {
+		return nil, nil
+	}
+
+	if k8sLifecycleHandler.Exec != nil {
+		p.logger.Info("exec container lifecycle is not supported, skipping")
```
Comment (shayrybak) [inline internal/provider/workload_translation.go:329]: probably should be a warning level

**Review by shayrybak — CHANGES_REQUESTED** [review]

File: internal/provider/workload_translation.go (hunk 2 above)
Comment (rolikoff) [reply internal/provider/workload_translation.go]: Updated here and in a few more places.

File: README.md [hunk 3]
```diff
@@ -27,10 +27,11 @@ Here are the specifications for each of the available instance types:
...
 In addition, the workloads created on the StackPath platform will not have network access to the Kubernetes API or any pods running in nodes that aren't in the virtual kubelet provider.
 - **Pod name length**. The provider is subject to the limitations of StackPath's workload slugs, which are limited to 63 characters. The provider constructs the slug by concatenating the namespace with the pod name separated by a dash. It is important to ensure that this string does not exceed 63 characters, as exceeding this limit will prevent the pod from being created.
+- **Lifecycle's PostStart**. The `postStart` hook in the lifecycle should only be used when the container is expected to start immediately. However, in most cases, this hook fails when using `httpGet` and `tcpSocket` action handlers because it is triggered immediately after the container is created, and many container applications do not start listening at that time. In a future release, we plan to support the `exec` action handler, which should be useful for the `postStart` hook.
```
[the remark this answers is not in the document]
Comment (rolikoff) [reply README.md]: Updated.

**Comment by shayrybak** [comment]
@rolikoff can we please add all the new fields that are supported to the readme?

**Review by shayrybak — APPROVED** [review]

**Pull request merged**